hadoop-cloud/src/hadoop-3/main/scala/org/apache/spark/internal/io/cloud/PathOutputCommitProtocol.scala

```diff
@@ -0,0 +1,281 @@
+   * to commit, even when a store has listing inconsistency (i.e. Amazon S3),
+   * there is no risk of that affecting the operation.
+   * Similarly, because the superclass's implementation of this method uses
+   * a UUID in the source name, eventual consistency on an update will not be
+   * an issue.
+   * There is a very small risk that Amazon S3's brief caching of 404s
+   * can cause any probes for the file to fail if the task is committed
+   * immediately after the file was created.
+   *
+   * @param taskContext task
+   * @param absoluteDir destination dir
+   * @param ext         extension
+   * @return an absolute path
+   */
+  override def newTaskTempFileAbsPath(
+    taskContext: TaskAttemptContext,
+    absoluteDir: String,
+    ext: String): String = {
+
+    val file = super.newTaskTempFileAbsPath(taskContext, absoluteDir, ext)
+    logInfo(s"Creating temporary file $file for absolute path for dir $absoluteDir")
+    file
+  }
+
+  /**
+   * Build a filename which is unique across all task events.
+   * It does not have to be consistent across multiple attempts of the same
+   * task or job.
+   *
+   * @param taskContext task context
+   * @param ext         extension
+   * @return a name for a file which must be unique across all task attempts
+   */
+  protected def buildFilename(
+    taskContext: TaskAttemptContext,
+    ext: String): String = {
+
+    // The file name looks like part-00000-2dd664f9-d2c4-4ffe-878f-c6c70c1fb0cb_00003-c000.parquet
+    // Note that %05d does not truncate the split number, so if we have more than 100000 tasks,
+    // the file name is fine and won't overflow.
+    val split = taskContext.getTaskAttemptID.getTaskID.getId
+    f"part-$split%05d-$jobId$ext"
+  }
+
+  override def setupJob(jobContext: JobContext): Unit = {
+    logDebug("setup job")
+    super.setupJob(jobContext)
+  }
+
+  override def commitJob(
+      jobContext: JobContext,
+      taskCommits: Seq[FileCommitProtocol.TaskCommitMessage]): Unit = {
+    logDebug(s"commit job with ${taskCommits.length} task commit message(s)")
+    super.commitJob(jobContext, taskCommits)
+  }
+
+  /**
+   * Abort the job; log and ignore any IO exception thrown.
+   * This is invariably invoked in an exception handler; raising
+   * an exception here will lose the root cause of the failure.
+   *
+   * @param jobContext job context
+   */
+  override def abortJob(jobContext: JobContext): Unit = {
+    try {
+      super.abortJob(jobContext)
+    } catch {
+      case e: IOException =>
+        logWarning("Abort job failed", e)
+    }
+  }
+
+  override def setupTask(taskContext: TaskAttemptContext): Unit = {
+    super.setupTask(taskContext)
+  }
+
+  override def commitTask(
+      taskContext: TaskAttemptContext): FileCommitProtocol.TaskCommitMessage = {
+      logDebug("Commit task")
+    super.commitTask(taskContext)
+  }
+
+  /**
+   * Abort the task; log and ignore any failure thrown.
```

Comment:
I understand the spirit, but if that's so important, why is it not done in the parent class? This feels like the wrong place for it.

Reply:
I think it should be in the parent. It surfaces with these committers more because 

1. the primary failure of an abort task in a classic cluster FS is that the cluster FS is unreachable, which is traumatic enough more things fail.
1. with alternative cloud stores and the like, there are different ways for things to go wrong. For example, the user issuing requests to AWS S3 doesn't have the "abort multipart upload" permission.

If you think it should be pulled up —which delivers consistency— then I'm happy to do this

Reply:
> If you think it should be pulled up

Yes, I think that's better. That way the reasoning is consistent - preserve the original exception, always - instead of different committers behaving in different ways.

Reply:
Were you planning on addressing this, or at least filing a follow-up bug?

Reply:
pulled up with separate handlers for committer abort and staging dir cleanup. Rationale: it may be that a committer fails for some reason (e.g. IAM permissions for abort) but delete() still works.